pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -101,6 +101,8 @@ public ResultTable submitAndReduce(long requestId, DispatchableSubPlan dispatcha
     } catch (Exception e) {
       cancel(requestId, dispatchableSubPlan);
       throw new RuntimeException("Error executing query: " + ExplainPlanPlanVisitor.explain(dispatchableSubPlan), e);
+    } catch (Throwable t) {
```

Comment:
should you just merge Throwable with Exception?

Reply:
When I add `cancel(requestId, dispatchableSubPlan);` it tries to make a grpc call again and end up throwing the  `NameResolver` error message. Do you suggest a nested try-catch around cancel as well then?

Reply:
1. when a Throwable is caught, it usually indicates series issues and cannot be recovered. 
2. netty handles the throwable thus the server didn't crash.

so here is my .02
1. don't need to explain the query, the throwable is serious enough that probably has nothing to do with the query itself.
2. log it and re-throw the throwable makes more sense to me instead of wrapping it back into a RTE

Reply:
Planning to move this catch clause to `MultiStageBrokerRequestHandler` instead and merge it with the catch exception there.